tools/cloud-build/purge_timedout.py

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env python3
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from google.cloud.devtools import cloudbuild_v1
+from google.cloud.devtools.cloudbuild_v1.types.cloudbuild import Build
+from google.cloud import storage
+from datetime import timezone, timedelta, datetime, date
+import glob
+import os
+import re
+import argparse
+import subprocess
+
+def run(project):
+    cb = cloudbuild_v1.services.cloud_build.CloudBuildClient()
+    req = cloudbuild_v1.ListBuildsRequest(
+        project_id=project, 
+        filter="status=TIMEOUT", 
+        page_size=1000
+    )
+    builds = cb.list_builds(req).builds
+    if not builds:
+        print("No timedout builds found this week.")
+        return
+
+    PDT = timezone(timedelta(hours=-7))
+    pastweek = datetime.combine(date.today(), datetime.min.time()).astimezone(PDT) - timedelta(days=7)
```

Comment:
I think `datetime.combine(date.today()...).astimezone(PDT)` only does the right thing if run from within the PDT (which is only active half the year)